integration-cli/docker_cli_update_unix_test.go

```diff
@@ -213,5 +213,16 @@ func (s *DockerSuite) TestUpdateStats(c *check.C) {
 	curMemLimit := getMemLimit(name)
 
 	c.Assert(preMemLimit, checker.Equals, curMemLimit)
+}
+
+func (s *DockerSuite) TestUpdateMemoryToZero(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	name := "test-update-memory"
+	dockerCmd(c, "create", "--name", name, "--memory", "50M", "busybox")
+	_, _, err := dockerCmdWithError("update", "--memory", "-1", name)
```

Comment:
It's not sufficient to update a created container, you should also update a started container.